token/jwt/map_claims.go

```diff
@@ -0,0 +1,154 @@
+package jwt
+
+import (
+	"crypto/subtle"
+	"encoding/json"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// MapClaims provides backwards compatible validations not available in `go-jose`.
+// It was taken from [here](https://github.com/dgrijalva/jwt-go/blob/master/map_claims.go).
+//
+// > Claims type that uses the map[string]interface{} for JSON decoding
+// This is the default claims type if you don't supply one
+type MapClaims map[string]interface{}
+
+// Compares the aud claim against cmp.
+// If required is false, this method will return true if the value matches or is unset
+func (m MapClaims) VerifyAudience(cmp string, req bool) bool {
+	aud, ok := m["aud"].([]string)
+	if !ok {
+		strAud, ok := m["aud"].(string)
+		if !ok {
+			return false
```

Comment:
If I read this correctly, this would fail when `aud` is not set and `req` is `false` which is contrary to what `If required is false, this method will return true if the value matches or is unset` says.

I think a test for this would be good!

Reply:
Good catch, agree, actually the error is in the forked `https://github.com/form3tech-oss/jwt-go/blob/v3.2.1/map_claims.go` and not in `https://github.com/dgrijalva/jwt-go/blob/v3.2.0/map_claims.go`. I will double check and update the PR with the fix and some tests with a new commit so you do not have to review all again.

Reply:
I updated with latest `map_claims.go` from `form3tech-oss/jwt-go` which accepts `aud` as an array. I noted that repo provides `map_claims_test.go`, I copy them and added a test to check for the inconsistency that you found.